apps/hash-frontend/src/components/hooks/use-orgs-with-links.ts

```diff
@@ -0,0 +1,82 @@
+import { ApolloQueryResult, useQuery } from "@apollo/client";
+import { OrgProperties } from "@local/hash-isomorphic-utils/system-types/shared";
+import {
+  AccountGroupId,
+  Entity,
+  EntityRootType,
+  Subgraph,
+} from "@local/hash-subgraph";
+import { getRoots } from "@local/hash-subgraph/stdlib";
+import { useMemo } from "react";
+
+import {
+  QueryEntitiesQuery,
+  QueryEntitiesQueryVariables,
+} from "../../graphql/api-types.gen";
+import { queryEntitiesQuery } from "../../graphql/queries/knowledge/entity.queries";
+import { constructOrg, Org } from "../../lib/user-and-org";
+
+/**
+ * Retrieves a specific set of organizations, with their avatars and members populated
+ */
+export const useOrgsWithLinks = ({
+  orgAccountGroupIds,
+}: {
+  orgAccountGroupIds: AccountGroupId[];
+}): {
+  loading: boolean;
+  orgs?: Org[];
+  refetch: () => Promise<ApolloQueryResult<QueryEntitiesQuery>>;
+} => {
+  const { data, loading, refetch } = useQuery<
+    QueryEntitiesQuery,
+    QueryEntitiesQueryVariables
+  >(queryEntitiesQuery, {
+    variables: {
+      operation: {
+        multiFilter: {
+          filters: orgAccountGroupIds.map((entityId) => ({
+            field: ["metadata", "recordId", "uuid"],
+            operator: "EQUALS",
+            value: entityId,
```

Comment:
Should we try and refer to `entityUuid` explicitly in our code? I think it could lead to confusion otherwise.
```suggestion
          filters: orgAccountGroupIds.map((entityUuid) => ({
            field: ["metadata", "recordId", "uuid"],
            operator: "EQUALS",
            value: entityUuid,
```

Reply:
Done in https://github.com/hashintel/hash/pull/3191/commits/677831f0abf9e7f3eebb8d6d4243515633702627